controllers/machinehealthcheck_controller.go

```diff
@@ -347,7 +347,7 @@ func (r *MachineHealthCheckReconciler) patchHealthyTargets(ctx context.Context,
 			if obj.GetDeletionTimestamp() == nil {
 				// Issue a delete for remediation request.
 				if err := r.Client.Delete(ctx, obj); err != nil && !apierrors.IsNotFound(err) {
-					logger.Error(err, "failed to delete %v %q for Machine %q", obj.GroupVersionKind(), obj.GetName(), t.Machine.Name)
+					errList = append(errList, errors.Wrapf(err, "failed to delete %v %q for Machine %q", obj.GroupVersionKind(), obj.GetName(), t.Machine.Name))
```

Comment:
@goushicui Can you explain why you'd like this to return here? Did you identify some unexpected behaviour with the current version?

Reply:
If we did this, wouldn't we want to only append if error != notFound?

Reply:
yes, just a choice that waiting nornal change machine or node update  to triger next reconceil or return error to tiger controller error handler?

Reply:
> If we did this, wouldn't we want to only append if error != notFound?

I missed it too, but that check is already in l.349.